test/functional/ui/screen.lua

```diff
@@ -565,12 +565,12 @@ asynchronous (feed(), nvim_input()) and synchronous API calls.
   end
 end
 
-function Screen:sleep(ms)
+function Screen:sleep(ms, on_request)
   local function notification_cb(method, args)
     assert(method == 'redraw')
     self:_redraw(args)
   end
-  run_session(self._session, nil, notification_cb, nil, ms)
+  run_session(self._session, on_request, notification_cb, nil, ms)
```

Comment:
I like `on_foo` better, but since we already use `foo_cb` everywhere, shouldn't we name this `request_cb` as well (at least internally; for the `screen:expect` key could still be renamed `on_request`).